Panic in timer_manager.rs due to unwrap() on failed ID allocation

### Describe the bug

The kernel panics with called Option::unwrap() on a None value when a user process attempts to create a POSIX timer after the timer ID allocator has been exhausted.

Instead of handling the allocation failure gracefully (e.g., by returning an -EAGAIN error to the user), the kernel code at kernel/src/process/process/timer_manager.rs:182 appears to force an unwrap on the result of the ID allocation. When the maximum number of timers is reached and the allocator returns None, this unwrap causes the kernel to crash.

# https://github.com/asterinas/asterinas/blob/8ace76978eadd74dd2952866c269ac88b41c577b/kernel/src/process/process/timer_manager.rs#L182

### To Reproduce

1. Compile the program and run.

```
#include <stdio.h>
#include <stdlib.h>
#include <signal.h>
#include <time.h>
#include <unistd.h>
#include <errno.h>

#define MAX_TIMERS 65536

int main() {
    timer_t timers[MAX_TIMERS];
    struct sigevent sev;
    int i;
    
    // Set up sigevent for timer creation
    sev.sigev_notify = SIGEV_SIGNAL;
    sev.sigev_signo = SIGUSR1;
    sev.sigev_value.sival_ptr = NULL;
    
    printf("Attempting to exhaust POSIX timer ID allocator...\n");
    
    // Create timers in a loop to exhaust the ID allocator
    for (i = 0; i < MAX_TIMERS; i++) {
        if (timer_create(CLOCK_REALTIME, &sev, &timers[i]) == -1) {
            if (errno == ENOMEM || errno == EAGAIN) {
                printf("Failed to create timer %d (errno: %d). This may trigger the kernel panic.\n", i, errno);
                break;
            } else {
                perror("timer_create");
                exit(1);
            }
        }
        
        if (i % 1000 == 0) {
            printf("Created %d timers...\n", i);
        }
    }
    
    printf("Successfully created %d timers before failure.\n", i);
    
    // Try to create one more timer to trigger the panic
    printf("Attempting to create one more timer to trigger the panic...\n");
    timer_t extra_timer;
    if (timer_create(CLOCK_REALTIME, &sev, &extra_timer) == -1) {
        perror("Final timer_create failed");
    }
    
    // Keep the process alive to observe the kernel panic
    printf("Process sleeping. Check kernel logs for panic.\n");
    while (1) {
        sleep(10);
    }
    
    return 0;
}

```

### Environment

- Official docker asterinas/asterinas:0.17.0-20251217
- Asterinas version: main [Add unit tests for DMA `alloc` and `alloc_uninit()` · asterinas/asterinas@8ace769](https://github.com/asterinas/asterinas/commit/8ace76978eadd74dd2952866c269ac88b41c577b)

### Logs

```
./ex106_asterinas__kernel__src__process__process__timer_manager.rs_182_24
Attempting to exhaust POSIX timer ID allocator...
Created 0 timers...
Created 1000 timers...
Created 2000 timers...
Created 3000 timers...
Created 4000 timers...
Created 5000 timers...
Created 6000 timers...
Created 7000 timers...
Created 8000 timers...
Created 9000 timers...
[   360.099] ERROR: Uncaught panic:
        called `Option::unwrap()` on a `None` value
        at /root/asterinas/kernel/src/process/process/timer_manager.rs:182
        on CPU 0 by thread Some(Thread { task: (Weak), data: Any { .. }, is_exited: false, cpu_affinity: AtomicIdSet { bits: [1], phantom: PhantomData<ostd::cpu::id::CpuId> }, sched_attr: SchedAttr { policy: SchedPolicyState { kind: AtomicSchedPolicyKind(2), policy: UnsafeCell { .. } }, last_cpu: AtomicCpuId(0), real_time: RealTimeAttr { prio: 99, time_slice: 31488960 }, fair: FairAttr { weight: 1024, pending_weight: 0, vruntime: 1027028230 } } })
Printing stack trace:
   1: fn 0xffffffff88970160 - pc 0xffffffff889701ab / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2c730;

   2: fn 0xffffffff882187c0 - pc 0xffffffff88218fbd / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2c750;

   3: fn 0xffffffff8821a780 - pc 0xffffffff8821a78a / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2cb20;

   4: fn 0xffffffff8804b000 - pc 0xffffffff8804b00a / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2cb30;

   5: fn 0xffffffff88ae4210 - pc 0xffffffff88ae4256 / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2cb40;

   6: fn 0xffffffff88ae41e0 - pc 0xffffffff88ae4206 / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2cb80;

   7: fn 0xffffffff88aca360 - pc 0xffffffff88aca375 / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2cba0;

   8: fn 0xffffffff88139a10 - pc 0xffffffff88139a8d / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2cbb0;

   9: fn 0xffffffff885771e0 - pc 0xffffffff88577bcd / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2cc40;

  10: fn 0xffffffff885f89f0 - pc 0xffffffff8862157e / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd2d0f0;

  11: fn 0xffffffff885f6a50 - pc 0xffffffff885f6aea / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd404f0;

  12: fn 0xffffffff8807e920 - pc 0xffffffff8807f1d1 / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40650;

  13: fn 0xffffffff8807f390 - pc 0xffffffff8807f3ee / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40b40;

  14: fn 0xffffffff886617f0 - pc 0xffffffff8866181e / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40cf0;

  15: fn 0xffffffff88661720 - pc 0xffffffff88661766 / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40d20;

  16: fn 0xffffffff885c94f0 - pc 0xffffffff885c951c / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40d90;

  17: fn 0xffffffff8813b520 - pc 0xffffffff8813b55f / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40dd0;

  18: fn 0xffffffff8807f350 - pc 0xffffffff8807f377 / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40f40;

  19: fn 0xffffffff8805f290 - pc 0xffffffff8805f2ad / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40f80;

  20: fn 0xffffffff88945e40 - pc 0xffffffff88945e65 / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40fa0;

  21: fn 0xffffffff889631a0 - pc 0xffffffff88963224 / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd40fd0;

  22: fn 0xffffffff8896df2d - pc 0xffffffff8896df36 / registers:

     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd41000;


     rax 0xffffdfffffd2c648; rdx 0xffffffff889a18a2; rcx  0x10000020100a6c5; rbx                0x0;
     rsi                0x0; rdi 0xffffffff88bc93f8; rbp                0x0; rsp 0xffffdfffffd41008;

OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) ostd::panic::print_stack_trace
          at /root/asterinas/ostd/src/panic.rs:113
(  2) aster_kernel::thread::oops::panic_handler
          at /root/asterinas/kernel/src/thread/oops.rs:130
(  3) __ostd_panic_handler
          at ??:?
(  4) __rustc::rust_begin_unwind
          at 5hffciyz3d33t9gfe7c8dzufi:?
(  5) core::panicking::panic_fmt
          at ??:?
(  6) core::panicking::panic
          at ??:?
(  7) core::option::unwrap_failed
          at ??:?
(  8) <core::option::Option<usize>>::unwrap
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/option.rs:1015
(  9) aster_kernel::syscall::timer_create::sys_timer_create
          at /root/asterinas/kernel/src/syscall/timer_create.rs:111
( 10) aster_kernel::syscall::arch::syscall_dispatch
          at /root/asterinas/kernel/src/syscall/mod.rs:217
( 11) aster_kernel::syscall::handle_syscall
          at /root/asterinas/kernel/src/syscall/mod.rs:374
( 12) aster_kernel::thread::task::create_new_user_task::{closure#0}
          at /root/asterinas/kernel/src/thread/task.rs:95
( 13) aster_kernel::thread::task::create_new_user_task::{closure#1}
          at /root/asterinas/kernel/src/thread/task.rs:126
( 14) unwinding::panicking::catch_unwind::do_call::<aster_kernel::thread::task::create_new_user_task::{closure#1}, ()>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panicking.rs:56
( 15) unwinding::panicking::catch_unwind::<unwinding::panic::RustPanic, (), aster_kernel::thread::task::create_new_user_task::{closure#1}>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panicking.rs:42
( 16) unwinding::panic::catch_unwind::<(), aster_kernel::thread::task::create_new_user_task::{closure#1}>
          at /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.8/src/panic.rs:87
( 17) aster_kernel::thread::oops::catch_panics_as_oops::<aster_kernel::thread::task::create_new_user_task::{closure#1}, ()>
          at /root/asterinas/kernel/src/thread/oops.rs:55
( 18) aster_kernel::thread::task::create_new_user_task::{closure#2}
          at /root/asterinas/kernel/src/thread/task.rs:131
( 19) <aster_kernel::thread::task::create_new_user_task::{closure#2} as core::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 20) <alloc::boxed::Box<dyn core::ops::function::FnOnce<(), Output = ()> + core::marker::Send> as core::ops::function::FnOnce<()>>::call_once
          at /root/.rustup/toolchains/nightly-2025-12-06-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2206
( 21) kernel_task_entry
          at /root/asterinas/ostd/src/task/mod.rs:196
( 22) kernel_task_entry_wrapper
          at ??:?
make: *** [Makefile:309: run_kernel] Error 1
```

--- a/kernel/src/process/process/timer_manager.rs
+++ b/kernel/src/process/process/timer_manager.rs
@@ -175,18 +175,18 @@ impl PosixTimerManager {
     }
 
     /// Adds a POSIX timer to the managed `posix_timers`, and allocate a timer ID for this timer.
-    /// Return the timer ID.
-    pub fn add_posix_timer(&self, posix_timer: Arc<Timer>) -> usize {
+    /// Return the timer ID, or `None` if allocation failed.
+    pub fn add_posix_timer(&self, posix_timer: Arc<Timer>) -> Option<usize> {
         let mut timers = self.posix_timers.lock();
         // Holding the lock of `posix_timers` is required to operate the `id_allocator`.
-        let timer_id = self.id_allocator.lock().alloc().unwrap();
-        if timers.len() < timer_id + 1 {
+        let timer_id = self.id_allocator.lock().alloc()?;
+        if timers.len() <= timer_id {
             timers.resize(timer_id + 1, None);
         }
         // The ID allocated is not used by any other timers so this index in `timers`
         // must be `None`.
         timers[timer_id] = Some(posix_timer);
-        timer_id
+        Some(timer_id)
     }
 
     /// Finds a POSIX timer by the input `timer_id`.

--- a/kernel/src/syscall/timer_create.rs
+++ b/kernel/src/syscall/timer_create.rs
@@ -108,7 +108,9 @@ pub fn sys_timer_create(
 
     let timer = create_timer(clockid, func, ctx)?;
 
-    let timer_id = current_process.timer_manager().add_posix_timer(timer);
+    let Some(timer_id) = current_process.timer_manager().add_posix_timer(timer) else {
+        return_errno_with_message!(Errno::EAGAIN, "timer IDs are exhausted");
+    };
     ctx.user_space().write_val(timer_id_addr, &timer_id)?;
     Ok(SyscallReturn::Return(0))
 }

